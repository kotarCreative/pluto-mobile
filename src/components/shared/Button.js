/**
 * Standard button component
 *
 * @format
 * @flow
 */

import React from 'react';
import { PureComponent } from 'react';
import { StyleSheet, Text, TouchableHighlight } from 'react-native';
import COLOR from '@/constants/colors'

// styles
const styles = StyleSheet.create({
  button: {
    padding: 5,
    borderRadius: 5,
    borderColor: COLOR.BLACK,
    borderWidth: 1
  },
  enabledButton: {
    backgroundColor: COLOR.BRAND_SECONDARY_HIGHLIGHT,
    borderWidth: 0
  },
  text: {
    color: COLOR.BLACK
  },
  enabledText: {
    color: COLOR.WHITE
  }
});

export class Button extends PureComponent<Props, State> {
  render() {
    return (
      <TouchableHighlight
        onPress={this.props.onPress}
        style={{ ...styles.button, ...!this.props.disabled ? styles.enabledButton : {}}}
        underlayColor={COLOR.BRAND_SECONDARY_HIGHLIGHT}
      >
        <Text style={{ ...styles.text, ...!this.props.disabled ? styles.enabledText : {}}}>{this.props.title}</Text>
      </TouchableHighlight>
    );
  }
}
