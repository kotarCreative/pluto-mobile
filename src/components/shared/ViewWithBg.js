/**
 * Standard background component
 *
 * @format
 * @flow
 */
import React, { PureComponent } from 'react'
import { Image, StyleSheet, View } from 'react-native'
import mainBg from '@/assets/images/sky-bg.png' //eslint-disable-line

// style
const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
  main: {
    position: 'absolute',
    top: 0,
    left: 0,
    flex: 1,
    height: '100%',
    width: '100%'
  },
  bg: {
    resizeMode: 'cover',
    height: '100%',
    width: '100%'
  },
  inner: {
    flex: 1,
    backgroundColor: 'transparent',
    justifyContent: 'center',
  }
})

export class ViewWithBg extends PureComponent<Props, State> {
  render() {
    return (
      <View
        style={styles.wrapper}
        {...this.props}
      >
        <View style={styles.main}>
          <Image
            style={styles.bg}
            source={mainBg}
          />
        </View>
        <View style={styles.inner}>
          {this.props.children}
        </View>
      </View>
    )
  }
}
