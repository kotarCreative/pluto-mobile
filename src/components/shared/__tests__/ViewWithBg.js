import React from 'react'
import renderer from 'react-test-renderer'

import { Button } from '../Button';
import { ViewWithBg } from '../ViewWithBg'

it('renders correctly with defaults', () => {
  const view = renderer
    .create(<ViewWithBg><Button title="Hello"/></ViewWithBg>)
    .toJSON()
  expect(view).toMatchSnapshot()
});
