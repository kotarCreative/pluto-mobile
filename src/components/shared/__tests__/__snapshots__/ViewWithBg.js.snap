// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/sky-bg@2x.png",
        }
      }
      style={
        Object {
          "height": "100%",
          "resizeMode": "cover",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E38C3B",
          "borderColor": "#000",
          "borderRadius": 5,
          "borderWidth": 0,
          "padding": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "textAlign": "center",
          }
        }
      >
        Hello
      </Text>
    </View>
  </View>
</View>
`;
